docs/platform/tutorials/image-processing/object-detection.md

```diff
@@ -1,67 +1,122 @@
-                .add_value("lat", timestamp.parameters["lat"].numeric_value) \
-                .add_value("lon", timestamp.parameters["lon"].numeric_value) \
-                .add_value("delta", delta) \
-                .write()
-    ```
+1. In the pipeline view, click on the object detection service tile.
 
-    1. Each time a new parameter data arrives, this callback is invoked.
-    2. Parameter data can be thought of as data in a tabular form. This code loops over all rows in the table.
-    3. The object detection model is called with the source image. It returns an annoted image, an array containing the ids of the types of objects detected (for example: ['bus', 'car', 'truck', 'car', 'car', 'person', 'car', 'car', 'person', 'car', 'car', 'car', 'person', 'car', 'person', 'person']), and the confidence of the detection.
-    4. The `delta` is a variable used to record how long it takes for the object detection. This is used as a measure of performance. 
-    5. `Counter` is a dictionary that object type counts, for example, `{'truck': 3, 'car': 3}`.
-    6. Timestamp TDB
-    7. Add the class ID (object type detected) and the count for that object type. 
-    8. The row is written out. The row includes the image binary, geolocation, and delta as a measure of performance for object detection.
+2. Click on the Logs tab, if not selected by default.
 
-[Part 4 - TfL video :material-arrow-right-circle:{ align=right }](connect-video-tfl.md)
+3. You can now see the log messages being produced by the service:
+
+    ![Object detection logs](../image-processing/images/object-detection-logs.png)
+
+    !!! tip
+
+        There is a pause button to allow you to pause the logs (see the screenshot). There is also a button you can use to download the logs for the service.
+
+There  also some tasks for you to carry out in the following sections.
+
+## 👩‍🔬 Lab - Examine the application code
+
+You now learn how to examine the code for the service. You may want to fix bugs in it, or otherwise improve the code for a service. Once the code is edited, you can use the `Redeploy` button the redeploy the service, even if it is already running.
+
+1. Click the panel indicated in the screenshot:
+
+    ![object detection code](../image-processing/images/object-detection-code.png){width=60%}
+
+    This takes you to the code view. You can view or edit the complete code for this service here. 
+    
+2. You could for example, make some changes, and then redeploy the service using the `Redeploy` button, or simply test your changes using the `Run` button.
+
+3. In the code view, click the `History` tab to see the complete revision history for changes to the code.
+
+### Task - Check the output message format for this service
+
+Using what you have learned in previous parts of this tutorial, check the format of the messages published by this service. 
+
+### Task - Determine the output topic name
```

Comment:
I'm a bit hesitant about this task. In my personal opinion its a deficit of the UI that you can't clearly see the topics in a "birds eye view" rather you have to hover over the arrows individually. What I mean is... it shouldn't be a "challenge" find out the connecting topics.

Reply:
Removed. I've made it into a tip earlier on.